src/Atmos/Model/gcminfo.jl

```diff
@@ -0,0 +1,187 @@
+export GCMModel, NoGCM, HadGEMVertical
+
+abstract type GCMModel end
+
+vars_state(::GCMModel, ::Prognostic, FT) = @vars()
+vars_state(::GCMModel, ::Auxiliary, FT) = @vars()
+vars_state(::GCMModel, ::Gradient, FT) = @vars()
+vars_state(::GCMModel, ::GradientFlux, FT) = @vars()
+vars_state(::GCMModel, ::UpwardIntegrals, FT) = @vars()
+vars_state(::GCMModel, ::DownwardIntegrals, FT) = @vars()
+
+function atmos_nodal_update_auxiliary_state!(
+    ::GCMModel,
+    ::AtmosModel,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+) end
+function integral_set_auxiliary_state!(
+    ::GCMModel,
+    integ::Vars,
+    state::Vars,
+    aux::Vars,
+) end
+function integral_load_auxiliary_state!(::GCMModel, aux::Vars, integ::Vars) end
+function reverse_integral_set_auxiliary_state!(
+    ::GCMModel,
+    integ::Vars,
+    state::Vars,
+    aux::Vars,
+) end
+function reverse_integral_load_auxiliary_state!(
+    ::GCMModel,
+    aux::Vars,
+    integ::Vars,
+) end
+function flux_radiation!(
+    ::GCMModel,
+    atmos::AtmosModel,
+    flux::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+) end
+@inline function atmos_nodal_update_auxiliary_state!(
+    gcminfo::GCMModel,
+    atmos::AtmosModel,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    nothing
+end
+
+function compute_gradient_argument!(
+    gcminfo::GCMModel,
+    transform::Vars,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    nothing
+end
+
+function compute_gradient_flux!(
+    gcminfo::GCMModel,
+    diffusive::Vars,
+    ∇transform::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    nothing
+end
+
+function flux_first_order_gcm!(
+    gcminfo::GCMModel,
+    atmos::AtmosModel,
+    flux::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    nothing
+end
+
+function flux_second_order!(
+    gcminfo::GCMModel,
+    flux::Grad,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    D_t,
+)
+    nothing
+end
+function flux_second_order!(gcminfo::GCMModel, flux::Grad, state::Vars, d_q_tot)
```

Comment:
Same, do we need this?

Reply:
We probably don't need any of these. Akshay and I will check and remove them if they are not needed.